lib/spectrum/request/record.rb

```diff
@@ -22,6 +22,13 @@ def initialize(request)
         @request = request
         if request.params[:source] == 'summon'
           @query = "#{@request.params['id_field']}:#{unfiltered_id(request)}"
+        elsif request.params[:source] == 'mirlyn'
+          id_field = @request.params['id_field']
```

Comment:
Is params['id_field'] always set? If not, you run the risk of sending a `nil`, which will in fact override the keyword argument in fetch. You may want/need to explicitly `id ||= 'id'` in the body of `Holdings#fetch` if you're not positive `nil` will never come in.

Reply:
The existing code seems to assume here that `params['id_field']` is always set (lines 24 and 33 here, or 24 and 26 in the `primo` branch). I also do have `request[:id_field] || 'id'` in the body of Request::Holdings#initialize, which is the `request` that is being passed in to `BibRecord#fetch`.